src/components/CommentInput.jsx

```diff
@@ -1,53 +1,84 @@
-// import { useEffect, useState } from "react";
-// import { getCommentsByArticleID, postCommentByArticleID } from "../api/api";
-
-// export const CommentInput = (params) => {
-//   const [comment, setComment] = useState("");
-//   const [loading, setLoading] = useState(true);
-//   const [commentSubmitted, setCommentSubmitted] = useState(false);
-
-//   const { id, onCommentPosted} = params;
-
-//   useEffect(() => {
-//   }, [commentSubmitted]);
-
-
-
-//   const handleSubmit = (e) => {
-//     e.preventDefault();
-
-//     setLoading(true);
-    
-//     const body = {
-//         username: "grumpy19",
-//       body: comment
-//     };
-//     console.log(id, body)
-//     postCommentByArticleID(id, body)
-//       .then((data) => {
-//         setLoading(false);
-//         setCommentSubmitted(true);
-//         onCommentPosted()
-//         setComment("")
-//       })
-
-//       .catch((err) => {
-//         setLoading(true);
-//         console.log(err);
-//       });
-//   };
-
-//   return (
-//     <div className="post-comment-card">
-//       <form className="comment-form" onSubmit={handleSubmit}>
-//         <textarea
-//           className="comment-input"
-//           placeholder="type comment here"
-//           value={comment}
-//           onChange={(e) => setComment(e.target.value)}
-//         />
-//         <button type="submit">Create Post</button>
-//       </form>
-//     </div>
-//   );
-// };
+import { useEffect, useState } from "react";
+import { getCommentsByArticleID, postCommentByArticleID } from "../api/api";
+
+export const CommentInput = (params) => {
+  const [comment, setComment] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [commentWarning, setCommentWarning] = useState(false);
+  const [commentEmpty, setCommentEmpty] = useState(false);
+
+  const { id, onCommentPosted } = params;
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    if (comment.length > 1000) {
+      setCommentWarning(true);
+      return;
+    }
+
+    if (comment.length === 0) {
+      setCommentEmpty(true);
+      return;
+    }
+
+    setLoading(true);
+    setError(false);
+
+    const body = {
+      username: "grumpy19",
+      body: comment,
+    };
+
+    postCommentByArticleID(id, body)
+      .then((data) => {
+        setLoading(false);
+        onCommentPosted(body);
```

Comment:
You'd want to use data, not body for this. 
i presume body is whats been typed in. Data will be the comment back from the db - it will have the comment_id on it that the database has given it. You'll need this later on to delete the comment.